src/main/java/slimeknights/tconstruct/library/recipe/casting/ContainerFillingRecipe.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.library.recipe.casting;
+
+import lombok.Getter;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.IItemProvider;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+public abstract class ContainerFillingRecipe implements ICastingRecipe {
+  @Getter
+  protected final IRecipeType<?> type;
+  @Getter
+  protected final ResourceLocation id;
+  @Getter
+  protected final String group;
+  @Getter
+  protected final int fluidAmount;
+  @Getter
+  protected final IItemProvider result;
+  @Getter
+  protected final boolean consumed = true;
+  protected final boolean switchSlots = false;
+  @Getter
+  protected final int coolingTime = 5;
+
+  public ContainerFillingRecipe(IRecipeType<?> typeIn, ResourceLocation idIn, String groupIn, int fluidAmount, IItemProvider resultIn) {
+    this.type = typeIn;
+    this.id = idIn;
+    this.group = groupIn;
+    this.fluidAmount = fluidAmount;
+    this.result = resultIn;
+  }
+
+  @Override
+  public int getFluidAmount(ICastingInventory inv) {
+    return this.fluidAmount;
+  }
+
+  @Override
+  public boolean switchSlots() {
+    return switchSlots;
+  }
+
+  @Override
+  public int getCoolingTime(ICastingInventory inv) {
+    return getCoolingTime();
+  }
+
+  @Override
+  public boolean matches(ICastingInventory inv, World worldIn) {
+    return inv.getStack().getItem() == this.result.asItem();
+  }
+
+  @Override
+  public ItemStack getRecipeOutput() {
+    return ItemStack.EMPTY;
+  }
```

Comment:
I suggest returning a stack of the empty container, so at least it has something. Probably also mark deprecated so people know this won't really work right.

Reply:
How does marking as deprecated work again?

Reply:
`@Deprecated` annotation. Can also throw in a java doc, `/** @deprecated use {@link whatever} */` or `/** @deprecated unused */`